bdns/dns_test.go

```diff
@@ -190,13 +188,31 @@ func serveLoopResolver(stopChan chan bool) chan bool {
 			fmt.Println(err)
 		}
 	}()
-	return waitChan
+}
+
+func pollServer() {
+	var conn *dns.Conn
+	timeout := time.Duration(time.Millisecond * 200)
+	tries := 25
+
+	for i := 0; conn == nil; i++ {
```

Comment:
Heh sorry, I wrote the first one then rushed to write out the second. If you could take a stab at the second version that would be awesome, it's better of a golang pattern to use in general and should make the code a bit easier to follow.